pyanaconda/geoloc.py

```diff
@@ -117,22 +117,100 @@
 from pyanaconda import constants
 from pyanaconda.threads import AnacondaThread, threadMgr
 from pyanaconda.timezone import get_preferred_timezone, is_valid_timezone
+from pyanaconda.flags import flags
 
+geolocation_enabled = False
 location_info_instance = None
 refresh_condition = threading.Condition()
 refresh_in_progress = False
 
 
-def init_geolocation(provider_id=constants.GEOLOC_DEFAULT_PROVIDER):
+def set_geolocation_enabled(options_override, install_class_override):
+    """Check if geolocation can be used during this installation run.
```

Comment:
I don't think that the name of the function `set_geolocation_enabled()` is doing what the name is telling us. It is even more clear when you look on the first row of the docstring: 
"**Check** if geolocation can be used during this installation run." 
There is no word about setting anything but more importantly there isn't a word about checking in the function name.

You could rename this function to `check_and_set_geolocation_enabled()` however then you have the `and` word which clearly telling us this function having a side effect and it also violate SRP principle.

I would recommend you to rewrite this to two functions. One will do just a checking and the second one will be a simple setter. It will be easier for any future modifications and bug hunting.